### DISCUSSION TITLE
Can we assume that server-side sockets are listening in connect()?

### DISCUSSION THREAD

**Discussion by ytrewq271828** [ANLAB-KAIST/KENSv3#192] [Project Q&A 2. Transport Layer (TCP)]
Hello.

When implementing connect(), is it okay to assume that the server-side socket, which is associated with sockaddr type parameter of the connect() call, is in the listening state? (listen() call is successfully called)?

Thanks in advance.

**Comment by Pusnow** [comment]
No, you also need to implement simultaneous connects, which do not enter LISTEN state.


https://github.com/ANLAB-KAIST/KENSv3/blob/5bf97d4d228d6a6bcb8a2fdc1f50abe9c075c701/Documentation/rfc/rfc793.txt#L2077-L2093
